mqtt/mqtt-edgehub/tests/authorization.rs

```diff
@@ -0,0 +1,105 @@
+use futures_util::StreamExt;
+
+use mqtt3::{
+    proto::ClientId, proto::PacketIdentifier, proto::QoS, proto::Subscribe, proto::SubscribeTo,
+};
+use mqtt_broker::{auth::DenyAll, BrokerBuilder};
+use mqtt_broker_tests_util::{
+    client::TestClientBuilder,
+    packet_stream::PacketStream,
+    server::{start_server, DummyAuthenticator},
+};
+use mqtt_edgehub::{
+    auth::EdgeHubAuthorizer, auth::IdentityUpdate, command::AuthorizedIdentitiesCommand,
+};
+
+mod common;
+
+/// Scenario:
+// create broker
+// create command handler
+// publish authorization update
+// connect authorized client and subscribe
+// publish authorization update with client removed
+// verify client has disconnected
+#[tokio::test]
+async fn disconnect_client_on_auth_update() {
+    // Start broker with DummyAuthorizer that allows everything from CommandHandler and $edgeHub,
+    // but otherwise passes authorization along to EdgeHubAuthorizer
+    let broker = BrokerBuilder::default()
+        .with_authorizer(common::DummyAuthorizer::new(EdgeHubAuthorizer::new(
+            DenyAll,
+        )))
+        .build();
+    let broker_handle = broker.handle();
+
+    let server_handle = start_server(broker, DummyAuthenticator::anonymous());
+
+    // start command handler with AuthorizedIdentitiesCommand
+    let command = AuthorizedIdentitiesCommand::new(&broker_handle);
+    let (command_handler_shutdown_handle, join_handle) =
+        common::start_command_handler(common::TEST_SERVER_ADDRESS.to_string(), command)
```

Comment:
I know you used disconnect test as an example, and I think there is a problem here and in the disconnect test.

if you look at `mqtt_broker_test_util::server::run` you'll find a code that increment a port number on every server start. This is done to avoid multiple tests running in parallel to compete for the same port number. Using hardcoded `TEST_SERVER_ADDRESS` will lead to unexpected behaviour.

Quick fix can be like this, since we don't care much about public/private transports in tests. @dmolokanov, let me know if you have better suggestions.
```suggestion
        common::start_command_handler(server_handle.address(), command)
```

Reply:
looks good